File: 0049-group-anagrams/0049-group-anagrams.cpp
class Solution {
public:
    vector<vector<string>> groupAnagrams(vector<string>& strs) {
        unordered_map<string,vector<string>> dictStr;
        
        for (const auto& s: strs){
            string tmpS=s;
            sort(tmpS.begin(), tmpS.end());
            dictStr[tmpS].push_back(s);
        }
        
        vector<vector<string>> result;
        for (const auto& pair: dictStr){
            result.push_back(pair.second);
        }
        
        return result;
    }
};

File: 0049-group-anagrams/0049-group-anagrams.py
"""
anagram끼리 묶어서 return

정렬해서 그 key 안에 넣기


"""

class Solution:
    def groupAnagrams(self, strs: List[str]) -> List[List[str]]:
        dictStr={}
        for s in strs:
            tmpS=''.join(sorted(s))
            if tmpS not in dictStr:
                dictStr[tmpS]=[]
            dictStr[tmpS].append(s)
        return dictStr.values()